pkg/workloadlearning/handle.go

```diff
@@ -217,4 +314,408 @@ func (handle *Handle) SaveTableReadCostMetrics(metrics map[ast.CIStr]*TableReadC
+		return "", ""
+	}
+	for _, child := range op.Children {
+		childOT, err := extractOperatorTypeFromName(child.Name)
+		if err != nil {
+			return "", ""
+		}
+		if childOT == plancodec.TypeIndexRangeScan || childOT == plancodec.TypeIndexFullScan {
+			dbName, tableName := extractTableNameFromAccessObject(child.AccessObjects[0])
+			return dbName, tableName
+		}
+		// Recursive search for the children layer
+		dbName, tableName := extractTableNameFromIndexScan(child)
+		if dbName != "" && tableName != "" {
+			return dbName, tableName
+		}
+	}
+	return "", ""
+}
+
+func extractPartialMetricsFromChildrenIndexMerge(op *tipb.ExplainOperator) ([]*TableReadCostMetrics, error) {
+	if len(op.Children) == 0 {
+		return nil, nil
+	}
+	// Find the layer of IndexxxxScan
+	children0OperatorType, err := extractOperatorTypeFromName(op.Children[0].Name)
+	if err != nil {
+		return nil, err
+	}
+	if children0OperatorType == plancodec.TypeIndexRangeScan || children0OperatorType == plancodec.TypeIndexFullScan {
+		result := make([]*TableReadCostMetrics, 0, len(op.Children))
+		for _, child := range op.Children {
+			dbName, tableName := extractTableNameFromAccessObject(child.AccessObjects[0])
+			if dbName == "" || tableName == "" {
+				return nil, fmt.Errorf("failed to get table name from children index xxx operator, operator name: %s", child.Name)
+			}
+			scanTime, err := extractScanTimeFromExecutionInfo(child)
+			if err != nil {
+				return nil, fmt.Errorf("failed to extract scan time from execution info, operator name: %s", child.Name)
+			}
+			partialMetric := &TableReadCostMetrics{
+				DbName:        ast.NewCIStr(dbName),
+				TableName:     ast.NewCIStr(tableName),
+				TableScanTime: scanTime,
+			}
+			result = append(result, partialMetric)
+		}
+		return result, nil
+	} else {
+		// Recursively find the children layer
+		var err error
+		var result []*TableReadCostMetrics
+		for _, child := range op.Children {
+			result, err = extractPartialMetricsFromChildrenIndexMerge(child)
+			if len(result) != 0 {
+				return result, nil
+			}
+		}
+		return nil, err
+	}
+}
+
+func extractOperatorTypeFromName(name string) (string, error) {
+	names := strings.Split(name, "_")
+	if len(names) != 2 {
+		return "", fmt.Errorf("failed to extract operator type from operator name: %s", name)
+	}
+	return names[0], nil
+}
+
+func extractScanTimeFromExecutionInfo(op *tipb.ExplainOperator) (time.Duration, error) {
+	var scanTime time.Duration
+	var err error
+	if len(op.RootBasicExecInfo) > 0 {
+		scanTime, err = extractScanTimeFromString(op.RootBasicExecInfo)
+	}
+	if scanTime == 0 && len(op.RootGroupExecInfo) > 0 {
+		// try groupExecInfo instead
+		scanTime, err = extractScanTimeFromString(op.RootGroupExecInfo[0])
+	}
+	if scanTime == 0 && len(op.CopExecInfo) > 0 {
+		// try copExecInfo instead
+		scanTime, err = extractScanTimeFromString(op.CopExecInfo)
+	}
+	return scanTime, err
+}
+
+func extractScanTimeFromString(input string) (time.Duration, error) {
```

Comment:
Better to add an example of the `input` as comments here. Otherwise it's hard to know what the `input` look like.